src/software/simulation/standalone_simulator_main.cpp

```diff
@@ -1,20 +1,106 @@
+        DynamicParameters->getNetworkConfig()->NetworkInterface()->value();
+    int yellow_team_channel =
+        DynamicParameters->getStandaloneSimulatorConfig()->YellowTeamChannel()->value();
+    std::string yellow_team_ip =
+        std::string(MULTICAST_CHANNELS[yellow_team_channel]) + "%" + network_interface;
+    int blue_team_channel =
+        DynamicParameters->getStandaloneSimulatorConfig()->BlueTeamChannel()->value();
+    std::string blue_team_ip =
+        std::string(MULTICAST_CHANNELS[blue_team_channel]) + "%" + network_interface;
+
     StandaloneSimulatorGUIWrapper standalone_simulator_gui_wrapper(argc, argv);
-    ThreadedSimulator simulator = ThreadedSimulator(Field::createSSLDivisionBField());
+    ThreadedSimulator simulator(Field::createSSLDivisionBField());
+
+    ThreadedProtoMulticastSender<SSL_WrapperPacket> wrapper_packet_sender(
+        SSL_VISION_DEFAULT_MULTICAST_ADDRESS, SSL_VISION_MULTICAST_PORT);
     simulator.registerOnSSLWrapperPacketReadyCallback(
-        [&standalone_simulator_gui_wrapper](SSL_WrapperPacket packet) {
+        [&standalone_simulator_gui_wrapper,
+         &wrapper_packet_sender](SSL_WrapperPacket packet) {
+            wrapper_packet_sender.sendProto(packet);
             standalone_simulator_gui_wrapper.onValueReceived(packet);
         });
 
-    simulator.setBallState(BallState(Point(1, 1), Vector(2, 1.2)));
+    ThreadedProtoMulticastListener<PrimitiveSetMsg> yellow_team_primitive_listener(
+        yellow_team_ip, PRIMITIVE_PORT, [&simulator](PrimitiveSetMsg msg) {
+            for (const auto& primitive : msg.robot_primitives())
+            {
+                RobotId id = primitive.first;
+                auto [primitive_index, primitive_params] =
+                    decodePrimitiveMsg(primitive.second);
+                simulator.setYellowRobotPrimitive(id, primitive_index, primitive_params);
+            }
+        });
+    ThreadedProtoMulticastListener<PrimitiveSetMsg> blue_team_primitive_listener(
+        blue_team_ip, PRIMITIVE_PORT, [&simulator](PrimitiveSetMsg msg) {
+            for (const auto& primitive : msg.robot_primitives())
+            {
+                RobotId id = primitive.first;
+                auto [primitive_index, primitive_params] =
+                    decodePrimitiveMsg(primitive.second);
+                simulator.setBlueRobotPrimitive(id, primitive_index, primitive_params);
+            }
+        });
+
+    simulator.setBallState(BallState(Point(0, 0), Vector(0, 0)));
+
+    RobotState blue_robot_state1(Point(-1, 0), Vector(0, 0), Angle::zero(),
+                                 AngularVelocity::zero());
+    RobotState blue_robot_state2(Point(-2, 1), Vector(0, 0), Angle::quarter(),
+                                 AngularVelocity::zero());
+    std::vector<RobotStateWithId> blue_robot_states = {
+        RobotStateWithId{.id = 1, .robot_state = blue_robot_state1},
+        RobotStateWithId{.id = 2, .robot_state = blue_robot_state2},
```

Comment:
Why two robots?

Reply:
arbitrary, just wanted to put some robots in there to show stuff off